src/net.cpp

```diff
@@ -2230,7 +2231,38 @@ NodeId CConnman::GetNewNodeId()
     return nLastNodeId.fetch_add(1, std::memory_order_relaxed);
 }
 
-bool CConnman::Start(CScheduler& scheduler, std::string& strNodeError, Options connOptions)
+
+bool CConnman::Bind(const CService &addr, unsigned int flags) {
+    if (!(flags & BF_EXPLICIT) && IsLimited(addr))
+        return false;
+    std::string strError;
+    if (!BindListenPort(addr, strError, (flags & BF_WHITELIST) != 0)) {
+        if ((flags & BF_REPORT_ERROR) && clientInterface) {
+            clientInterface->ThreadSafeMessageBox(strError, "", CClientUIInterface::MSG_ERROR);
+        }
+        return false;
+    }
+    return true;
+}
+
+bool CConnman::InitBinds(const std::vector<std::pair<CService, Options::BindType> >& binds) {
+    bool fBound = false;
+    for (const auto& addrBind : binds) {
+        unsigned int flags = BF_EXPLICIT | BF_REPORT_ERROR;
+        if (addrBind.second == Options::BindType::WhiteBind)
+            flags |= BF_WHITELIST;
+        fBound |= Bind(addrBind.first, flags);
+    }
+    if (binds.empty()) {
```

Comment:
Looks like there's no way to run without listening now.

Sorry for not being more clear here: https://github.com/bitcoin/bitcoin/pull/10496#discussion_r120199183. I was suggesting that we shouldn't get into InitBinds in the first place if binding isn't allowed.

We'll either need to change this so that its caller uses the ugly global as before:
``` c++
    if(fListen && !InitBinds(...)) {
        ...
    }
```
or add a flag for fallback connections, which InitBinds() only uses if nothing else is already bound.